main.cpp

```diff
@@ -19,13 +19,30 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
-
+    CastCallBack(Rewriter& rewriter): _rewriter(rewriter) {};
+    
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        auto *cast = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        auto type = Lexer::getSourceText(
+                                CharSourceRange::getTokenRange(

```

Comment:
Please descrbe the difference between the use of `getCharRange()` and `getTokenRange()` and why `getTokenRange()` is used here.

Reply:
`getCharRange()` returns range specifying the starting character and ending character by the specified locations.
`getTokenRange()` returns range that specify the start of the range and the start of the last token of the range.

The `getCharRange()` method returns a string within the given range, while the `getTokenRange()` method will return the entire token within the range.

Both options are acceptable, but because they return different values, the arguments passed must be different.

Reply:
Sounds right!
To see the difference between these two methods, you can run the following code:
  
```cpp
  // Difference between getCharRange and getTokenRange 
  auto charRange = Lexer::getAsCharRange(
      CharSourceRange::getCharRange(node->getLParenLoc().getLocWithOffset(1),
      node->getRParenLoc().getLocWithOffset(-2)),
      SM, Result.Context->getLangOpts());
  
  auto tokenRange = Lexer::getAsCharRange(
      CharSourceRange::getTokenRange(node->getLParenLoc().getLocWithOffset(1),
      node->getRParenLoc().getLocWithOffset(-2)),
      SM, Result.Context->getLangOpts());
  
  llvm::outs() << "\n char range start: " << charRange.getBegin().printToString(SM) << ",\n char range end: " << charRange.getEnd().printToString(SM) << "\n";
  llvm::outs() << "\n token range start: " << tokenRange.getBegin().printToString(SM) << ",\n token range
 end: " << tokenRange.getEnd().printToString(SM) << "\n";
```  
  
  The output will be as follows:
  
```bash
   char range start: /UNN-C-Style-Checker/test/test.cpp:3:14,
   char range end: /UNN-C-Style-Checker/test/test.cpp:3:15
  
   token range start: /UNN-C-Style-Checker/test/test.cpp
:3:14,
   token range end: /UNN-C-Style-Checker/test/test.cpp:3:17
```  
  For the following example:
 ```cpp
  int main() {
      float foo;
      int i = (int)     foo ;
            // i - 15th item in line
            //   t - 17th item in line  
      return 0;
  }
  ```